### PR TITLE
New Debian container

### PR DESCRIPTION
**Pull request by cfelipesp**
Added  `debian` folder to `warewulf-node-images` with three files in it:

- Containerfile
- excludes
- container_exit.sh

It is a contribution to issue [#867](https://github.com/hpcng/warewulf/issues/867#).

### CODE REVIEW

**Comment by anderbubble** [comment]
This is great, @cfelipesp!

Could you integrate this into https://github.com/hpcng/warewulf-node-images/blob/main/.github/workflows/container-publish.yml as well?

And can you sign-off on your commit to indicate your attestation to https://github.com/hpcng/warewulf-node-images/blob/main/DCO.md?

File: debian/Containerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+FROM debian
+
+
+# squelsh apt-get
+ENV DEBIAN_FRONTEND noninteractive
+ENV RUNLEVEL 1
```
[the remark this answers is not in the document]
Comment (cfelipesp) [reply debian/Containerfile:6]: Well pointed, @anderbubble. Thanks!

Maybe we could just make it simple and neat, without any DEBIAN_FRONTEND  or RUNLEVEL. What do you think?

[the remark this answers is not in the document]
Comment (cfelipesp) [reply debian/Containerfile:6]: Like this:

``` dockerfile
FROM debian:12.0


# ----- install vital packages -----
RUN apt-get update && apt-get install -y --no-install-recommends \
	kmod \
	systemd-sysv \
	dbus \
	openssh-client \
	openssh-server \
	isc-dhcp-client \
	pciutils \
	strace \
	nfs-common \
	ethtool\
	ifupdown \
	linux-image-amd64 \
	ifmetric \
	netbase && \
	rm -rf /var/lib/apt/lists/*


COPY excludes /etc/warewulf/
COPY container_exit.sh /etc/warewulf/

CMD [ "/bin/echo", "-e", \
      "This image is intended to be used with the Warewulf cluster management and", \
      "\nprovisioning system.", \
      "\n", \
      "\nFor more information about Warewulf, visit https://warewulf.org" ]
```

File: debian/Containerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
+FROM debian
```
[the remark this answers is not in the document]
Comment (cfelipesp) [reply debian/Containerfile:1]: ok. I will change it to 

```Dockerfile
FROM debian:12.0
```

File: debian/Containerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (anderbubble) [reply debian/Containerfile:6]: If that works, no complaint from me. I haven't done scripted Debian installs in a while; I assumed if you were setting them that they needed to be set. :)

**Comment by cfelipesp** [comment]
> And can you sign-off on your commit to indicate your attestation to https://github.com/hpcng/warewulf-node-images/blob/main/DCO.md?

I'm sorry. I'm new to git and github. Only now I read I should have made all my commits using `commit -s` . Is there anything I can do now to sign-off on them?  Or should I just make sure I sign-off the new commits from now on? I'm sorry...

**Comment by anderbubble** [comment]
@cfelipesp I apologize for my absence here.

You can fix sign past commits with `git rebase --signoff`. Check out `git help rebase` for more info, and let me know if you have further questions. Or you can tell me what name / email address to use here and I'll sign them off for you.

Meanwhile, I'll get testing this, and I look forward to getting them merged in.

**Comment by anderbubble** [comment]
Do be advised, I've rebased this onto main; so you'll need to `pull` changes back to your machine if you want to continue working from the current branch.

**Comment by anderbubble** [comment]
Glorious. It appears to be working.

```
[janderson@wwctl1 debian-node-image]$ sudo wwctl node set wwnode1 --container debian-12.0
Are you sure you want to modify 1 nodes(s): y
? Are you sure you want to modify 1 nodes(s)? [y/N] y█

[janderson@wwctl1 debian-node-image]$ sudo wwctl overlay build
Building system overlays for wwnode1: [wwinit]
WARN   : context(system) and overlays([wwinit]) specified: prioritizing context(system)
Created image for overlay wwnode1/[wwinit]: /var/lib/warewulf/provision/overlays/wwnode1/__SYSTEM__.img
Compressed image for overlay wwnode1/[wwinit]: /var/lib/warewulf/provision/overlays/wwnode1/__SYSTEM__.img.gz
Building runtime overlays for wwnode1: [generic]
WARN   : context(runtime) and overlays([generic]) specified: prioritizing context(runtime)
Created image for overlay wwnode1/[generic]: /var/lib/warewulf/provision/overlays/wwnode1/__RUNTIME__.img
Compressed image for overlay wwnode1/[generic]: /var/lib/warewulf/provision/overlays/wwnode1/__RUNTIME__.img.gz

[janderson@wwctl1 debian-node-image]$ sudo ssh root@wwnode1
Linux wwnode1 6.1.0-14-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.64-1 (2023-11-30) x86_64

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.

root@wwnode1:~# cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

**Review by anderbubble — CHANGES_REQUESTED** [review]
@cfelipesp I squashed your commits together, and added you to the contributors file. I think this is ready to merge., except that I still need to get a signoff on it. Just let me know here for the record and I'll do it.

edit: looks like I spoke too soon. The container built for me locally, but isn't building in the GH action. I'll look into why.

**Comment by anderbubble** [comment]
```
#11 16.05 E: Unable to locate package linux-image-amd64
```

**Comment by anderbubble** [comment]
I updated the container files to install the correct kernel dependent on the intended architecture, and it's working now.

**Comment by cfelipesp** [comment]
> @cfelipesp I apologize for my absence here.
> 
> You can fix sign past commits with `git rebase --signoff`. Check out `git help rebase` for more info, and let me know if you have further questions. Or you can tell me what name / email address to use here and I'll sign them off for you.
> 
> Meanwhile, I'll get testing this, and I look forward to getting them merged in.

Hi, @anderbubble. I hope this is still in time, my e-mail is felpin1@gmail.com . Could you please sign them off for me, then? I'm afraid I could make a mess if I tried to do that myself. Thanks

**Comment by anderbubble** [comment]
@cfelipesp it's merged. Thanks for contributing!

**Comment by cfelipesp** [comment]
> @cfelipesp it's merged. Thanks for contributing!

My pleasure, @anderbubble ! 
Thank you for tutoring me in my first contribution to an open source project. And specially for being so patient with me. I have learned immensely.

**Pull request merged**